### ISSUE TITLE
nextflow nextflow/23.04.2 and tower.yml

### ISSUE THREAD

**Issue by jIskCoder** [nextflow-io/nextflow#4080]
### Expected behavior and actual behavior

Inlatest version trying tower.yml as described here https://seqera.io/blog/best-practices-for-deploying-pipelines-with-nextflow-tower/

gave a java error 

```
ERROR ~ 'com.fasterxml.jackson.core.io.ContentReference com.fasterxml.jackson.dataformat.yaml.YAMLFactory._createContentReference(java.lang.Object)'

 -- Check '.nextflow.log' file for details
```

Example tower.yml that didn't work

```
reports:
  **/multiqc_report.html:
    display : "MultiQC report"
```

```
reports:
  "**/multiqc_report.html":
    display : "MultiQC report"

```

**Comment by pditommaso** [comment]
Please provide the `.nextflow.log`

**Comment by jIskCoder** [comment]
`July-05 17:59:21.139 [main] DEBUG nextflow.cli.Launcher - $> /stornext/System/data/tools/nextflow/nextflow-23.04.2/nextflow-23.04.2-all run main.nf -profile milton -c config/fromfastq_nextflow.config
July-05 17:59:21.555 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.04.2
July-05 17:59:21.577 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/home/users/allstaff/iskander.j/.nextflow/plugins; core-plugins: nf-amazon@1.16.2,nf-azure@1.0.1,nf-codecommit@0.1.4,nf-console@1.0.5,nf-ga4gh@1.0.5,nf-google@1.7.3,nf-tower@1.5.12,nf-wave@0.8.3
July-05 17:59:21.594 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
July-05 17:59:21.595 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
July-05 17:59:21.599 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
July-05 17:59:21.609 [main] INFO  org.pf4j.AbstractPluginManager - No plugins
July-05 17:59:21.624 [main] DEBUG nextflow.config.ConfigBuilder - Found config home: /home/users/allstaff/iskander.j/.nextflow/config
July-05 17:59:21.625 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: /vast/projects/RCP/PPP_NF/nf-qc-pipe/nextflow.config
July-05 17:59:21.627 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /vast/projects/RCP/PPP_NF/nf-qc-pipe/config/fromfastq_nextflow.config
July-05 17:59:21.627 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/users/allstaff/iskander.j/.nextflow/config
July-05 17:59:21.628 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /vast/projects/RCP/PPP_NF/nf-qc-pipe/nextflow.config
July-05 17:59:21.628 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /vast/projects/RCP/PPP_NF/nf-qc-pipe/config/fromfastq_nextflow.config
July-05 17:59:21.650 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `milton`
July-05 17:59:22.510 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `milton`
July-05 17:59:22.874 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `milton`
July-05 17:59:23.080 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [debug, conda, milton]
July-05 17:59:23.139 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default
July-05 17:59:23.148 [main] INFO  nextflow.cli.CmdRun - Launching `main.nf` [silly_meucci] DSL2 - revision: 85d1fef957
July-05 17:59:23.149 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[]
July-05 17:59:23.150 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-tower@1.5.12]
July-05 17:59:23.150 [main] DEBUG nextflow.plugin.PluginsFacade - Plugin 'start' is not required in self-contained mode -- ignoring for plugin: nf-tower
July-05 17:59:23.167 [main] DEBUG nextflow.secret.LocalSecretsProvider - Secrets store: /home/users/allstaff/iskander.j/.nextflow/secrets/store.json
July-05 17:59:23.171 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@2668c286] - activable => nextflow.secret.LocalSecretsProvider@2668c286
July-05 17:59:23.270 [main] DEBUG nextflow.Session - Session UUID: 735e220c-bcd6-4262-ab7f-756a777ebc67
July-05 17:59:23.270 [main] DEBUG nextflow.Session - Run name: silly_meucci
July-05 17:59:23.271 [main] DEBUG nextflow.Session - Executor pool size: 2
July-05 17:59:23.282 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=10; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false
July-05 17:59:23.287 [main] DEBUG nextflow.config.ConfigBuilder - Found config home: /home/users/allstaff/iskander.j/.nextflow/config
July-05 17:59:23.287 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: /vast/projects/RCP/PPP_NF/nf-qc-pipe/nextflow.config
July-05 17:59:23.287 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /vast/projects/RCP/PPP_NF/nf-qc-pipe/config/fromfastq_nextflow.config
July-05 17:59:23.287 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/users/allstaff/iskander.j/.nextflow/config
July-05 17:59:23.288 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /vast/projects/RCP/PPP_NF/nf-qc-pipe/nextflow.config
July-05 17:59:23.288 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /vast/projects/RCP/PPP_NF/nf-qc-pipe/config/fromfastq_nextflow.config
July-05 17:59:23.288 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `milton`
July-05 17:59:23.344 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `milton`
July-05 17:59:23.560 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `milton`
July-05 17:59:23.682 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [debug, conda, milton]
July-05 17:59:23.751 [main] DEBUG nextflow.cli.CmdRun - 
  Version: 23.04.2 build 5871
  Created: 08-06-2023 09:33 UTC (19:33 AEDT)
  System: Linux 3.10.0-1160.88.1.el7.x86_64
  Runtime: Groovy 3.0.16 on OpenJDK 64-Bit Server VM 17.0.2+8-86
  Encoding: UTF-8 (UTF-8)
  Process: 31143@slurm-login01.hpc.wehi.edu.au [10.11.1.119]
  CPUs: 2 - Mem: 8 GB (1.1 GB) - Swap: 0 (0)
July-05 17:59:23.791 [main] DEBUG nextflow.Session - Work-dir: /vast/scratch/users/iskander.j/nextflow/work [nfs]
July-05 17:59:23.791 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: /vast/projects/RCP/PPP_NF/nf-qc-pipe/bin
July-05 17:59:23.837 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, AwsBatchExecutor, GoogleBatchExecutor]
July-05 17:59:23.850 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
July-05 17:59:23.867 [main] DEBUG nextflow.Session - Observer factory: TowerFactory
July-05 17:59:24.076 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory
July-05 17:59:24.088 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 3; maxThreads: 1000
July-05 17:59:24.201 [main] DEBUG nextflow.Session - Session start
July-05 17:59:24.216 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://tower.services.biocommons.org.au/api; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250
July-05 17:59:24.689 [main] ERROR nextflow.cli.Launcher - @unknown
java.lang.NoSuchMethodError: 'com.fasterxml.jackson.core.io.ContentReference com.fasterxml.jackson.dataformat.yaml.YAMLFactory._createContentReference(java.lang.Object)'
	at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:413)
	at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:15)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3561)
	at org.apache.groovy.yaml.util.YamlConverter.convertYamlToJson(YamlConverter.java:42)
	at groovy.yaml.YamlSlurper.parse(YamlSlurper.java:61)
	at groovy.yaml.YamlSlurper.parse(YamlSlurper.java:71)
	at groovy.yaml.YamlSlurper.parse(YamlSlurper.java:92)
	at io.seqera.tower.plugin.TowerReports.parseReportEntries(TowerReports.groovy:155)
	at io.seqera.tower.plugin.TowerReports.flowCreate(TowerReports.groovy:65)
	at io.seqera.tower.plugin.TowerClient.onFlowCreate(TowerClient.groovy:295)
	at nextflow.Session$_notifyFlowCreate_closure23.doCall(Session.groovy:1054)
	at nextflow.Session$_notifyFlowCreate_closure23.call(Session.groovy)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2357)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2342)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2383)
	at nextflow.Session.notifyFlowCreate(Session.groovy:1054)
	at nextflow.Session.start(Session.groovy:517)
	at nextflow.script.ScriptRunner.execute(ScriptRunner.groovy:125)
	at nextflow.cli.CmdRun.run(CmdRun.groovy:368)
	at nextflow.cli.Launcher.run(Launcher.groovy:494)
	at nextflow.cli.Launcher.main(Launcher.groovy:653)
`

**Comment by pditommaso** [comment]
Adding the full stack trace for reference 

```
July-05 17:59:24.689 [main] ERROR nextflow.cli.Launcher - @unknown
java.lang.NoSuchMethodError: 'com.fasterxml.jackson.core.io.ContentReference com.fasterxml.jackson.dataformat.yaml.YAMLFactory._createContentReference(java.lang.Object)'
	at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:413)
	at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:15)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3561)
	at org.apache.groovy.yaml.util.YamlConverter.convertYamlToJson(YamlConverter.java:42)
	at groovy.yaml.YamlSlurper.parse(YamlSlurper.java:61)
	at groovy.yaml.YamlSlurper.parse(YamlSlurper.java:71)
	at groovy.yaml.YamlSlurper.parse(YamlSlurper.java:92)
	at io.seqera.tower.plugin.TowerReports.parseReportEntries(TowerReports.groovy:155)
	at io.seqera.tower.plugin.TowerReports.flowCreate(TowerReports.groovy:65)
	at io.seqera.tower.plugin.TowerClient.onFlowCreate(TowerClient.groovy:295)
	at nextflow.Session$_notifyFlowCreate_closure23.doCall(Session.groovy:1054)
	at nextflow.Session$_notifyFlowCreate_closure23.call(Session.groovy)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2357)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2342)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2383)
	at nextflow.Session.notifyFlowCreate(Session.groovy:1054)
	at nextflow.Session.start(Session.groovy:517)
	at nextflow.script.ScriptRunner.execute(ScriptRunner.groovy:125)
	at nextflow.cli.CmdRun.run(CmdRun.groovy:368)
	at nextflow.cli.Launcher.run(Launcher.groovy:494)
	at nextflow.cli.Launcher.main(Launcher.groovy:653)
```

It looks like a problem with snakemake library

**Comment by jIskCoder** [comment]
it works fine on nextflow v22 but not v23

**Comment by jIskCoder** [comment]
Another issue I found is that tower.yml works fine when launching the pipeline for Tower GUI. If I have a running through cmd, no reports are created. No errors with v22 but no reports are visible on tower GUI although files have been created (.tsv and .HTML).

**Comment by bentsherman** [comment]
I tested with Nextflow 23.04.2. The first example fails because `**/multiqc_report.html` isn't quoted. But the second example worked for me.

As for the reports not showing up in Tower, that is a current limitation of monitoring-only Tower runs, it does not detect the reports. I think the Tower team is aware of this limitation.

**Comment by bentsherman** [comment]
@pditommaso I finally figured out why I could never reproduce this error. It only happens with the all release:
```
July-05 17:59:21.139 [main] DEBUG nextflow.cli.Launcher - $> /stornext/System/data/tools/nextflow/nextflow-23.04.2/nextflow-23.04.2-all run main.nf -profile milton -c config/fromfastq_nextflow.config
```

**Comment by pditommaso** [comment]
Even with a recent one?

**Comment by bentsherman** [comment]
Good point, it does not happen with 23.05.0-edge or 23.10.1. My guess is that it was fixed by this commit: 003e1f70
